zerver/lib/retention.py

```diff
@@ -153,31 +177,31 @@ def delete_expired_attachments(realm: Realm) -> None:
     )
     attachments_to_remove.delete()
 
+def move_expired_to_archive() -> Dict[str, Dict[str, List[int]]]:
+    archived_data_info = {}  # type: Dict[str, Dict[str, List[int]]]
+    for realm in Realm.objects.filter(message_retention_days__isnull=False).order_by("id"):
+        msg_ids = move_expired_messages_to_archive(realm)
+        usermsg_ids = move_expired_user_messages_to_archive(msg_ids)
+        move_expired_models_with_message_key_to_archive(msg_ids)
+        move_expired_attachments_to_archive(realm, msg_ids)
+        move_expired_attachments_message_rows_to_archive(realm, msg_ids)
 
-def clean_unused_messages() -> None:
-    unused_messages = Message.objects.filter(
-        usermessage__isnull=True, id__in=ArchivedMessage.objects.all()
-    )
-    unused_messages.delete()
+        archived_data_info[realm.string_id] = {
```

Comment:
Do we want to use `realm.id`, rather than `realm.string_id`, here?  Generally numeric IDs are more robust, so unless it's important for debugging/test readability, I'd prefer that model.

Reply:
Switched to using numeric id, there wasn't any particular reason for string_ids.